dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultDataQueryService.java

```diff
@@ -234,9 +234,11 @@ public DimensionalObject getDimension( String dimension, List<String> items, Eve
 
     @Override
     public DimensionalObject getDimension( String dimension, List<String> items, Date relativePeriodDate,
-        List<OrganisationUnit> userOrgUnits, boolean allowNull, IdScheme inputIdScheme )
+        List<OrganisationUnit> userOrgUnits, boolean allowNull, DisplayProperty displayProperty,
+        IdScheme inputIdScheme )
     {
-        return getDimension( dimension, items, relativePeriodDate, NAME, userOrgUnits, allowNull, inputIdScheme );
+        return getDimension( dimension, items, relativePeriodDate, displayProperty == null ? NAME : displayProperty,
```

Comment:
Would be good to avoid this in-line check. We should probably handle null further down in the method calls.

Reply:
Outdated due the requested changes to pass null down to the methods and handle it there